src/tir/transforms/lower_tvm_builtin.cc

```diff
@@ -140,6 +164,23 @@ class BuiltinLower : public StmtExprMutator {
       return StmtExprMutator::VisitStmt_(op);
     }
   }
+  Stmt VisitStmt_(const ForNode* op) final {
+    PrimExpr min = this->VisitExpr(op->min);
+    PrimExpr extent = this->VisitExpr(op->extent);
+    Stmt body;
+
+    if (op->kind == ForKind::kParallel) {
+      body = this->RealizeAlloca(op->body);
+    } else {
+      body = this->VisitStmt(op->body);
+    }
+    auto n = CopyOnWrite(op);
```

Comment:
if (min.same_as(op->min) && extent.same_as(op->extent) && body.same_as(body)) {
   return GetRef<Stmt>(op);
} else {
   // the following logic
}

Reply:
Done.